chore: bump to 0.91.0; update changelog (#15710)
rust-lang/cargo: src/doc/src/CHANGELOG.md

### Fixed

- Fixed potential deadlock in `CacheState::lock`
  [#15698](https://github.com/rust-lang/cargo/pull/15698)
  [#15699](https://github.com/rust-lang/cargo/pull/15699)
- Fixed the `--manifest-path` arg being ignored in `cargo fix`
  [#15633](https://github.com/rust-lang/cargo/pull/15633)
- When publishing, don't tell people to ctrl-c without knowing consequences.
  [#15632](https://github.com/rust-lang/cargo/pull/15632)
- Added missing `--offline` in shell completions.
  [#15623](https://github.com/rust-lang/cargo/pull/15623)
- cargo-credential-libsecret: load libsecret only once
  [#15295](https://github.com/rust-lang/cargo/pull/15295)

> - fix(fingerprint): When failing to find the mtime of a file, report an explicit reason rather than "stale; unknown reason".
>   [#15617](https://github.com/rust-lang/cargo/pull/15617)

Reviewer: ```suggestion
- When failing to find the mtime of a file for rebuild detection, report an explicit reason rather than "stale; unknown reason".
  [#15617](https://github.com/rust-lang/cargo/pull/15617)
```

- Fixed cargo add overwriting symlinked Cargo.toml files
  [#15281](https://github.com/rust-lang/cargo/pull/15281)

> - fix(vendor)!: vendor files with .rej/.orig suffix

Reviewer: ```suggestion
- Vendor files with .rej/.orig suffix
```

Also, this has a `!`, does this need to be highlighed in the with an emoji / put in the Changed section?

  `.rmeta` files.
  ([docs](https://doc.rust-lang.org/nightly/cargo/reference/unstable.html#no-embed-metadata))
  [#15378](https://github.com/rust-lang/cargo/pull/15378)
- 🔥 Plumb rustc `-Zhint-mostly-unused` flag through as a profile option
  ([docs](https://doc.rust-lang.org/nightly/cargo/reference/unstable.html#profile-hint-mostly-unused-option))
  [#15643](https://github.com/rust-lang/cargo/pull/15643)
- Added the "future" edition
  [#15595](https://github.com/rust-lang/cargo/pull/15595)
- Added `-Zfix-edition`
  [#15596](https://github.com/rust-lang/cargo/pull/15596)
- Added perma unstable `--compile-time-deps` option for `cargo build`
  [#15674](https://github.com/rust-lang/cargo/pull/15674)
- `-Zscript`: Make cargo script ignore workspaces.
  [#15496](https://github.com/rust-lang/cargo/pull/15496)
- `-Zpackage-workspace`: keep dev-dependencies if they have a version.
  [#15470](https://github.com/rust-lang/cargo/pull/15470)
- Added custom completer for `cargo remove <TAB>`
  [#15662](https://github.com/rust-lang/cargo/pull/15662)

> - test(publish): Improvements in prep for `-Zpackage-workspace` stabilization
>   [#15628](https://github.com/rust-lang/cargo/pull/15628)
> - fix(package): Allow packaging of self-cycles with `-Zpackage-workspace`
>   [#15626](https://github.com/rust-lang/cargo/pull/15626)
> - fix(trim-paths): remap all paths to `build.build-dir`
>   [#15614](https://github.com/rust-lang/cargo/pull/15614)
> - test(trim-paths): enable more tests for windows-msvc
>   [#15621](https://github.com/rust-lang/cargo/pull/15621)
> - fix(doc): pass `toolchain-shared-resources` to get doc styled for rustdoc-depinfo tracking
>   [#15605](https://github.com/rust-lang/cargo/pull/15605)
> - fix(embedded): Resolve multiple bugs in frontmatter parser for `-Zscript`
>   [#15573](https://github.com/rust-lang/cargo/pull/15573)
> - fix(toml): Remove workaround for rustc frontmatter support for `-Zscript`
>   [#15570](https://github.com/rust-lang/cargo/pull/15570)
> - Allow configuring arbitrary codegen backends
>   [#15562](https://github.com/rust-lang/cargo/pull/15562)
> - feat: skip `publish=false` pkg when publishing entire workspace for `-Zpackage-workspace`.
>   [#15525](https://github.com/rust-lang/cargo/pull/15525)
> - doc: Update instructions on using native-completions
>   [#15480](https://github.com/rust-lang/cargo/pull/15480)
> - fix(package): Skip registry check if its not needed with `-Zpackage-workspace`.
>   [#15629](https://github.com/rust-lang/cargo/pull/15629)

Reviewer: ```suggestion
- Test improvements in prep for `-Zpackage-workspace` stabilization
  [#15628](https://github.com/rust-lang/cargo/pull/15628)
- Allow packaging of self-cycles with `-Zpackage-workspace`
  [#15626](https://github.com/rust-lang/cargo/pull/15626)
- With trim-paths, remap all paths to `build.build-dir`
  [#15614](https://github.com/rust-lang/cargo/pull/15614)
- Enable more trim-paths tests for windows-msvc
  [#15621](https://github.com/rust-lang/cargo/pull/15621)
- Fix doc rebuild detection by passing `toolchain-shared-resources` to get doc styled for rustdoc-depinfo tracking
  [#15605](https://github.com/rust-lang/cargo/pull/15605)
- Resolve multiple bugs in frontmatter parser for `-Zscript`
  [#15573](https://github.com/rust-lang/cargo/pull/15573)
- Remove workaround for rustc frontmatter support for `-Zscript`
  [#15570](https://github.com/rust-lang/cargo/pull/15570)
- Allow configuring arbitrary codegen backends
  [#15562](https://github.com/rust-lang/cargo/pull/15562)
- skip `publish=false` pkg when publishing entire workspace for `-Zpackage-workspace`.
  [#15525](https://github.com/rust-lang/cargo/pull/15525)
- Update instructions on using native-completions
  [#15480](https://github.com/rust-lang/cargo/pull/15480)
- Skip registry check if its not needed with `-Zpackage-workspace`.
  [#15629](https://github.com/rust-lang/cargo/pull/15629)
```
